src/call.cpp

```diff
@@ -904,27 +895,13 @@ bool call::connect_socket_if_needed()
         }
 
         if (peripsocket) {
-            struct addrinfo * h ;
-            struct addrinfo   hints;
-            memset((char*)&hints, 0, sizeof(hints));
-            hints.ai_flags  = AI_PASSIVE;
-            hints.ai_family = PF_UNSPEC;
-            getaddrinfo(peripaddr,
-                        NULL,
-                        &hints,
-                        &h);
+            fill_sockaddr_from_ip(&saddr, peripaddr, use_ipv6);
+        } else {
             memcpy(&saddr,
-                   h->ai_addr,
-                   SOCK_ADDR_SIZE(
-                       _RCAST(struct sockaddr_storage *,h->ai_addr)));
-
-            if (use_ipv6) {
-                (_RCAST(struct sockaddr_in6 *, &saddr))->sin6_port = htons(local_port);
-            } else {
-                (_RCAST(struct sockaddr_in *, &saddr))->sin_port = htons(local_port);
-            }
+                    local_addr_storage->ai_addr,
+                    SOCK_ADDR_SIZE(
+                        _RCAST(struct sockaddr_storage *,local_addr_storage->ai_addr)));
```

Comment:
~~~c
memcpy(&saddr, local_addr_storage->ai_addr, sizeof(struct sockaddr_storage));
~~~

Is perfectly safe and simpler.